sdk/storage/Azure.Storage.Blobs/src/Generated/BlobRestClient.cs

```diff
@@ -9225,7 +9225,7 @@ public static partial class PageBlob
 
                 // Add request headers
                 _request.Headers.SetValue("x-ms-version", version);
-                if (prevSnapshotUrl != null) { _request.Headers.SetValue("x-ms-previous-snapshot-url", prevSnapshotUrl.ToString()); }
+                if (prevSnapshotUrl != null) { _request.Headers.SetValue("x-ms-previous-snapshot-url", prevSnapshotUrl.AbsoluteUri); }
                 if (range != null) { _request.Headers.SetValue("x-ms-range", range); }
```

Comment:
Nit - Might be worth adding a test for snapshots to validate we're treating this URI correctly as well.